src/main.js

```diff
@@ -46,12 +38,30 @@ if (args.dev) {
 //   https://cloud.google.com/nodejs/getting-started/logging-application-events
 const appWithLogging = express();
 
-appWithLogging.use(logging.middleware.requestLogger);
-appWithLogging.use(logging.middleware.errorLogger);
+appWithLogging.use(middleware.requestLogger);
+appWithLogging.use(middleware.errorLogger);
 appWithLogging.use(app);
 
-const server = appWithLogging.listen(port, () => {
-    const host = server.address().address;
-    const port = server.address().port;
-    logging.info("react-render-server running at http://%s:%s", host, port);
+const server = appWithLogging.listen(args.port, (err: ?Error) => {
+    if (server == null || err != null) {
+        logging.error(
+            `react-render-server appears not to have started: ${(err &&
+                err.message) ||
+                "Unknown error"}`,
+        );
+        return;
+    }
+
+    const address = server.address();
+    if (address == null || typeof address === "string") {
+        logging.warn(
+            "react-render-server may not have started properly: %s",
+            address,
+        );
+        return;
+    }
```

Comment:
It's amazing what additional cases you have to consider when flow gets involved. Turns out `address()` might return a string! And there was a possible error that we were ignoring, not to mention that `server` might not even be assigned.